tests/test_chmod_command.rs

```diff
@@ -0,0 +1,191 @@
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+
+    // Test chmodding a conflict
+    test_env.jj_cmd_success(&repo_path, &["chmod", "x", "file"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    Conflict:
+      Removing executable file with id df967b96a579e45a18b8251732d16804b2e56a55
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding executable file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+    test_env.jj_cmd_success(&repo_path, &["chmod", "n", "file"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    <<<<<<<
+    %%%%%%%
+    -base
+    +x
+    +++++++
+    n
+    >>>>>>>
+    "###);
+
+    // An error prevents `chmod` from making any changes.
+    // In this case, the failure with `nonexistent` prevents any changes to `file`.
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["chmod", "x", "nonexistent", "file"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: Can't find anything at `nonexistent`.
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    <<<<<<<
+    %%%%%%%
+    -base
+    +x
+    +++++++
+    n
+    >>>>>>>
+    "###);
+}
+
+#[test]
+fn test_chmod_file_dir_deletion_conflicts() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "file", &["base"], &[("file", "a\n")]);
+
+    create_commit(&test_env, &repo_path, "deletion", &["base"], &[]);
+    std::fs::remove_file(repo_path.join("file")).unwrap();
+
+    create_commit(&test_env, &repo_path, "dir", &["base"], &[]);
+    std::fs::remove_file(repo_path.join("file")).unwrap();
+    std::fs::create_dir(repo_path.join("file")).unwrap();
+    // Without a placeholder file, `jj` ignores an empty directory
+    std::fs::write(repo_path.join("file").join("placeholder"), "").unwrap();
+
+    // Create a file-dir conflict and a deletion-dir conflict
+    create_commit(&test_env, &repo_path, "file_dir", &["file", "dir"], &[]);
+    create_commit(
+        &test_env,
+        &repo_path,
+        "deletion_dir",
+        &["deletion", "dir"],
+        &[],
+    );
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    deletion_dir
+    ├─╮
+    │ │ ◉  file_dir
+    ╭───┤
+    ◉ │ │  dir
+    │ ◉ │  deletion
+    ├─╯ │
+    │   ◉  file
+    ├───╯
+    ◉  base
+    ◉
+    "###);
+
+    // The file-dir conflict can be chmod-ed
```

Comment:
This is an interesting case if we go with using matchers. Should `jj chmod file` update the directory recursively? We could error out and force the user to resolve the file/dir conflict before chmodding the result.

Reply:
> I feel like it should fail in this case; at least I can't imagine a situation where I'd want the other behavior.

But that's not what it currently does, it seems. So do you plan to update it so it becomes an error?

Reply:
Ah, sorry, for some reason I thought we were talking about a deletion-dir conflict or something like that.

I don't have a strong opinion about the file-dir conflict but, on second thought, I should probably change it to be an error. Will do!

Reply:
And done. :)